pymongo/common.py

```diff
@@ -871,6 +870,37 @@ def _ecoc_coll_name(encrypted_fields: Mapping[str, Any], name: str) -> Any:
 WRITE_CONCERN_OPTIONS = frozenset(["w", "wtimeout", "wtimeoutms", "fsync", "j", "journal"])
 
 
+def _get_timeout_details(client: MongoClient) -> dict[str, Any]:
+    details = {}
+    timeout = client.options.timeout
+    socket_timeout = client.options.pool_options.socket_timeout
+    connect_timeout = client.options.pool_options.connect_timeout
+    topology = client.topology_description.server_descriptions()
```

Comment:
This should be `client.topology_description` which is what we include in ServerSelectionTimeoutError: https://github.com/mongodb/mongo-python-driver/blob/6f4e617e6db6323b1da389b94a8a598d85e7c233/pymongo/topology.py#L259